### ISSUE TITLE
Include language server

### ISSUE THREAD

**Issue by prabhupraveen** [haskell/docker-haskell#116]
One needs to install GHCup and HLS on the host environment to be able to use HLS on VSCode. Also, while the VSCode plugin can manage automatic installation of the HLS, mapping Haskell version to supported HLS version has to be done manually on each project.

It would be great to include supported and matching HLS version in the docker image itself. By using VSCode devcontainers one can then launch docker container and get HLS features out of the box.

Assuming HLS is available within the image, one can easily launch a devcontainer in VSCode following these steps:

**Pre-requisites:**

- A working docker installation on the host (or docker plugin on VSCode)
- Enable Docker CLI for the non-root user account  (typically achieved by adding user to docker group) or be able to run VSCode as root (not recommended)
- A working VSCode installation with:
  a. Haskell and Haskell Syntax Highlighting extensions (with manual HLS option and no need to install GHCup)
  b. Devcontainer extension

**Project setup:**

- Create project folder structure as needed on the host.
- Inside the project root folder, create a folder called .devcontainer
- create a file called devcontainer.json inside the .devcontainer folder and add the following content:
```
{
    "image": "haskell:9-slim-buster",
  
    "customizations": {
      "vscode": {
        "extensions": ["haskell.haskell", "justusadam.language-haskell"]
      }
    }
}
```

Launch VSCode and open the project folder. VSCode will detect the devcontainer json and prompt reloading the project in devcontainer, and will launch it once accepted. One can now start developing in ghc with syntax highlighting and other HLS features.

I have managed to get this working on my ubuntu desktop. However, there are some observations:

- HLS debian binary is not available for aarach64, so adding HLS to Dockerfile will break aarch64 image build. Can be fixed by adding HLS aarch64 debian binary to the repository
- HLS binaries include support for multiple GHC versions. By modifying HLS make file, ghc version specific binary may be installed in the image. This will make development environment consistent and also significantly reduce image size.

I can submit a PR for Dockerfile, attached a working sample [here](https://github.com/haskell/docker-haskell/files/13816704/Dockerfile.zip)

**Comment by prabhupraveen** [comment]
I can see that this was already discussed and closed in #76. Hence closing this.

**Comment by prabhupraveen** [comment]
... Although, this solution did not involve installing GHCup in the container, but followed similar steps as installing stack, cabal and Haskell by directly downloading and installing HLS, hence a much cleaner solution, and I am of the opinion that including HLS (without GHCup) in the container still a good idea.
